apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -161,6 +172,14 @@ router.get(
         res.locals.start = groupInfo.start;
         res.locals.rolesInfo = groupInfo.rolesInfo;
         res.locals.used_join_code = req.body.used_join_code;
+
+        if (groupConfig.has_roles) {
+          const result = await groupAssessmentHelper.getAssessmentPermissions(
+            res.locals.assessment.id,
+            res.locals.user.user_id,
+          );
+          res.locals.canViewRoleTable = result.can_assign_roles_at_start;
+        }
```

Comment:
Was it a bug that this wasn't here before?

Reply:
Yes, exams did not have the role table in place.